### PR TITLE
Make k6 wait for loki to finish pushing before stopping

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
This is dependant on #1691

File: cmd/root.go
Comment (imiric) [inline cmd/root.go:208]: Not sure if we need a timeout if `^C` can stop this, no?

File: log/loki.go
Comment (imiric) [inline log/loki.go:292]: Since this is the cutoff time, is the second to deal with time sync issues and get some last events? Should it be equal to the push period, so maybe `h.pushPeriod` here instead?

**Review by imiric — COMMENTED** [review]
LGTM, just some questions and nitpicks.

File: cmd/root.go
Comment (MStoykov) [reply cmd/root.go:147]: I just moved this around ... this was written by @na-- a while ago https://github.com/loadimpact/k6/commit/a9cc1309e4fc8317ab105dd474d56740b1f0a197  :wink:  I don't think having this in an issue will help anyone, if you start working on it, maybe make one and we can discuss it but in general I consider this a refactoring and having refactoring issues given the amount of issues we have that IMO we will never actually work on .. so they are basically "ideas" makes the issues hard to use so ... yeah, I don't think this needs to be added to an issue. 

I myself have added issues (mostly refactoring ones) that again are hardly unlikely for us to ever work on ... and even if we did the fact that there is an issue would've not helped in any way.

On related note, I have a branch with 1 more commit where I get EVEN more of the global variables like noColor in a struct and get it passed around, but that will have to wait for v0.30.0 maybe I can try this then

File: log/loki.go
Comment (MStoykov) [reply log/loki.go:292]: yeah ... the basic idea is that ... as far as I can see you can call time.Now() in two parallel goroutines ... and you won't exactly get ... sequential times even if in reality they were ... so I added some time to get "hopefully" everything. 

I don't think that `h.pushPeriod` is better ... if anything I am more for `time.Hour` and a comment explaining why it is done at all (this took me 1 hour to figure out why I wasn't getting the last messages even though I was waiting for them and around 10 fmt.Printlns)

File: cmd/root.go
Comment (MStoykov) [reply cmd/root.go:208]: I guess not, and also the question is ... how big of timeout? I don't want to add one more configuration so :shrug: 
cc @na-- opinion ?

File: cmd/root.go
Comment (MStoykov) [reply cmd/root.go:73]: I think I meant to write 
```
This is to get all fields for the main/root k6 command struct
```
but I eat half the words ? I remember that this strut was 4 things in the begining so :shrug:

File: cmd/root.go
Comment (na--) [reply cmd/root.go:208]: I think this timeout should be handled by each logger individually, so it's easier to configure. And, unless I'm mistaken, the loki logger sort of has that in https://github.com/loadimpact/k6/blob/5091636dbf920aadb51d52d53f7d43fd30049dd6/log/loki.go#L97

That said, it won't hurt to add another hardcoded 5 second max timeout here, with a warning message to the fallback logger, just in case...

**Review by na-- — APPROVED** [review]
LGTM, though I'd prefer if we have the global timeout as well... And I'm not sure I like the `rootCommand` struct, but I guess it's better than globals and we're restricted by the structure imposed from cobra, so it's fine for now...

File: cmd/root.go
Comment (MStoykov) [reply cmd/root.go:102]: Hm ... that is a good idea

File: cmd/root.go
Comment (MStoykov) [reply cmd/root.go:208]: Okay, I am gonna add global timeout

**Review by na-- — APPROVED** [review]

**Review by imiric — APPROVED** [review]
The 1s delay to get the last messages seems a bit arbitrary, and maybe we should increase it or consider more robust solutions, but we can come back to it if needed, so LGTM.

**Pull request merged**
